### PR TITLE
Prevent bad settings files from crashing Semgrep

### PR DESCRIPTION
**Pull request by nbrahms**
Instead, just emit a warning and fall back to a default format.

Fixes PA-580.

PR checklist:
- [x] Documentation is up-to-date
- [x] Changelog is up-to-date
- [x] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

**Review by emjin — COMMENTED** [review]
Do we want to also output the settings file that fails? I would like to see what state the file gets in

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
   consistent with other use of ... (#4082), so `o. ... .foo()` will now
   also match just `o.foo()`.
 - Config files with only a comment give bad error message (#3773)
+- Prevent bad settings files from crashing Semgrep (#4164)
```
Comment (brendongo) [inline CHANGELOG.md]: Wrong place in changelog

**Comment by aryx** [comment]
And do we know why the error appeared randomly? The first time I was making a PR I got the error, and after 'rerun-all-jobs' it disappeared.

**Comment by nbrahms** [comment]
> And do we know why the error appeared randomly? The first time I was making a PR I got the error, and after 'rerun-all-jobs' it disappeared.

No clue why this happens TBH. I suspect it's a Python versioning / typing thing, but haven't been able to reproduce.

File: CHANGELOG.md (hunk 1 above)
Comment (nbrahms) [reply CHANGELOG.md]: ugh bad merge resolution

**Review by emjin — APPROVED** [review]

**Pull request merged**